### ISSUE TITLE
Constraints are limited to two options

### ISSUE THREAD

**Issue by jasonbreimer** [ocsf/ocsf-schema#870]
Currently constraints are limited to two options; `at_least_one` and `just_one`.

However I think we need an `all` option. This came up when working on the new kb_article object and it was suggested that several constraints should be applied. I believe we need this change to allow several constraints which would result in more flexibility.

https://github.com/ocsf/ocsf-docs/blob/592255d693ff12e3c4d987291e4b0ad0813d2799/Understanding%20OCSF.md
```json
 "constraints": {
   "at_least_one": [
     "ip",
     "mac",
     "name",
     "hostname"
   ]
 }
```

```json
"constraints": {
   "just_one": [
     "privileges",
     "group"
   ]
 }
}
```

**Comment by pagbabian-splunk** [comment]
I agree and also observed that these two options don't cover the case here.  However `all` is the same as making each attribute `Required` rather than `Recommended`.  The problem with other options like `at_least_two` is what if you want `at_least_three` etc.  We might want a more complex constraint, such as `at_least_M` with another parameter for `M` < `N`.

**Comment by mikeradka** [comment]
I align with Paul's comment, the only thing I would add is we may want to check to see if there are any server changes necessary for this.
